internal/blockdb/migrate.go

```diff
@@ -80,7 +84,21 @@ ON CONFLICT(git_sha) DO UPDATE SET git_sha=git_sha`, nowRFC3339(), gitSha)
 		return fmt.Errorf("create table tx: %w", err)
 	}
 
-	_, err = db.Exec(`DROP VIEW IF EXISTS v_tx_flattened`)
+	_, err = db.Exec(`ALTER TABLE chain ADD COLUMN chain_type TEXT NOT NULL check(length(chain_type) > 0) DEFAULT "cosmos"`)
+	if errIgnoreDuplicateColumn(err, "chain_type") != nil {
```

Comment:
Sqlite does NOT have an `alter column`. If postgres, I would `alter column chain_type drop default` after adding the column.

I think the default of `cosmos` will be fine for ibctest but if anyone has issue, happy to hear alternatives. Otherwise, an idempotent migration WITH constraints like `not null` becomes tricky. The contraints take priority imo.

Reply:
I assume that at this point in time, we are safe to expect any users to delete and recreate their database. Therefore my loosely held preference is that we just  change the `CREATE TABLE` call, and if anyone gets a weird error today, the recommendation is to delete and start over.

Moreover, we don't yet have proper migration tests in place. That is, we don't have a previous version of a database, populated with data, that we migrate to the current version. 

We don't need to decide just yet, but I prefer a firm commitment one way or the other -- we are never migrating so you always have to start fresh when we change the schema; or we are guaranteeing (within reason) that **all** your old data will continue to be available as ibctest evolves. Right now, I don't think the collected data is valuable enough to justify the latter.

Reply:
Github isn't letting me comment on your previous comment. ¯_(ツ)_/¯

Moreover, we don't yet have proper migration tests in place.
This is hard to pull off in practice. I've never worked at a place that did this well if at all. There's always bad and surprising data that you never accounted for. At previous companies, the strategy was if migrations failed, you did a quick rollback of the application. But that's in the classic web server context; it's different for embedded databases.

In this case, I think committing to the strategy of we delete (or backup) and start over is the way to go.

I'll attempt a PR at ☝️ strategy first.